documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,501 @@
-You can add hard goals through Cruise Control deployment configuration.
-You also have main, default, and user-provided goals that fit into each of these categories.  
-+
---
-* *Hard goals* are preset and must be satisfied for an optimization proposal to be successful.
-* *Soft goals* do not need to be satisfied for an optimization proposal to be successful. 
-They can be set aside if it means that all hard goals are met.  
-* *Main goals* are inherited from Cruise Control. Some are preset as hard goals.
-Main goals are used in optimization proposals by default.  
-* *Default goals* are the same as the main goals by default. 
-You can specify your own set of default goals.  
-* *User-provided goals* are a subset of default goals that are configured for generating a specific optimization proposal. 
---
-
-Optimization proposals::  Optimization proposals comprise the goals you want to achieve from a rebalance. 
-You generate an optimization proposal to create a summary of proposed changes and the results that are possible with the rebalance.
-The goals are assessed in a specific order of priority.
-You can then choose to approve or reject the proposal.
-You can reject the proposal to run it again with an adjusted set of goals.  
-+
-You can generate an optimization proposal in one of three modes.
-+
---
-* *`full`* is the default mode and runs a full rebalance.
-* *`add-brokers`* is the mode you use after adding brokers when scaling up a Kafka cluster.  
-* *`remove-brokers`* is the mode you use before removing brokers when scaling down a Kafka cluster.
---
-
-Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+Cruise Control comprises four main components: 
+
+Load Monitor:: Load Monitor collects the metrics and analyzes cluster workload data.
+Analyzer:: Analyzer generates optimization proposals based on collected data and configured goals.
+Anomaly Detector:: Anomaly Detector identifies and reports irregularities in cluster behavior.
+Executor:: Executor applies approved optimization proposals to the cluster.
+
+Cruise Control also provides a REST API for client interactions, which Strimzi uses to support these features:
+
+* Generating optimization proposals from optimization goals
+* Rebalancing a Kafka cluster based on an optimization proposal
+
+NOTE: Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+
+== Optimization goals
+
+Optimization goals define objectives for rebalancing, such as distributing topic replicas evenly across brokers. 
+
+They are categorized as follows:
+
+* *Hard goals* are preset and mandatory for a proposal to succeed.
+* *Soft goals* are optional and can be set aside if hard goals are met.
+* *Main goals* are inherited from Cruise Control, some preset as hard goals, used by default.
+* *Default goals* are the same as main goals by default but customizable.
```

Comment:
As the name suggests, the Analyzer's `default.goals` is what it's used by default when requesting a proposal, and it is equal to the Analyzer's `goals` if the user does not set `default.goals`. 

The `goals` property is customizable in case the user wants to restrict the supported goals to a subset of what CruiseControl provides by default. This means that, if a goal is not present in Analyzer's `goals`, the it cannot be used in Analyzer's `default.goals` and proposal's `goals`.

The rest of the documentation seems to match my understanding. Do you think we can improve the wording a bit here?

Reply:
Thanks. I've tried to make this clearer and keep succinct.